### PR TITLE
handle back button presses + refactor

### PR DESCRIPTION
**Pull request by Arkadyus**
This PR handles pressing the back button on the phone on any activity (Billsharer activities not tested as trying to open them crashes the app)
Every activity is now in the panels package
It also refactors the panels package into smaller packages (it feels cleaner to me but feel free to comment if you have a different idea)
This PR also updates the build and manifest in order to implement the google maps private key

### CODE REVIEW

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/LoginActivityTest.java [hunk 1]
```diff
@@ -55,4 +65,14 @@ public void discoverButtonIsEnabled() {
...
 
+    @Test
+    public void backPressLeavesApp() {
+        // Closing the app throws NoActivityResumedException, so we make the test fail if nothing was thrown
+        Intents.init();
+        try {
+            pressBack();
+            fail("Should have thrown NoActivityResumedException");
+        } catch (NoActivityResumedException expected) { }
+        Intents.release();
```
Comment (Geno22) [inline app/src/androidTest/java/com/github/houseorganizer/houseorganizer/LoginActivityTest.java:76]: Isn't it better to put the realease in a finally block ? I am not sure how it works though

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/MainScreenActivityTest.java [hunk 2]
```diff
@@ -150,6 +157,18 @@ public void calendarMenuFiresIntent() {
...
+
+    @Test
+    public void backPressLeavesApp() {
+        // Closing the app throws NoActivityResumedException, so we make the test fail if nothing was thrown
+        Intents.init();
+        try {
+            pressBack();
+            fail("Should have thrown NoActivityResumedException");
+        } catch (NoActivityResumedException expected) { }
+        Intents.release();
```
Comment (Geno22) [inline app/src/androidTest/java/com/github/houseorganizer/houseorganizer/MainScreenActivityTest.java:170]: Same as above

File: app/src/main/java/com/github/houseorganizer/houseorganizer/panels/login/LoginActivity.java [hunk 3]
```diff
@@ -126,4 +126,13 @@ private void firebaseAuthWithGoogle(String idToken) {
                 task -> manageTask(task, "firebaseAuthWithGoogle")
         );
     }
+
+    @Override
+    public void onBackPressed() {
+        // Leave the app instead of going to MainActivity
+        Intent homeIntent = new Intent(Intent.ACTION_MAIN);
+        homeIntent.addCategory(Intent.CATEGORY_HOME);
+        homeIntent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
```
Comment (Geno22) [inline app/src/main/java/com/github/houseorganizer/houseorganizer/panels/login/LoginActivity.java:135]: What does this flag do ? I am curious xD

**Review by Geno22 — CHANGES_REQUESTED** [review]
Great nice refactor pretty much nothing to say just a comment ^^

File: app/src/androidTest/java/com/github/houseorganizer/houseorganizer/LoginActivityTest.java (hunk 1 above)
Comment (Arkadyus) [reply app/src/androidTest/java/com/github/houseorganizer/houseorganizer/LoginActivityTest.java:76]: actually i think the safest way was to put them in the @after and the Intents.init() in the @before, ill move them there

File: app/src/main/java/com/github/houseorganizer/houseorganizer/panels/login/LoginActivity.java (hunk 3 above)
Comment (Geno22) [reply app/src/main/java/com/github/houseorganizer/houseorganizer/panels/login/LoginActivity.java:135]: Great thanks !

**Review by Geno22 — APPROVED** [review]
Thanks for the changes !

**Review by Oxydeas — APPROVED** [review]
Clean addition. Thanks for fixing this and for the general refactor of the files (65 files changed tho!)

**Pull request merged**